services/drupal/web/modules/custom/epa_core/epa_core.deploy.php

```diff
@@ -433,18 +433,157 @@ function epa_core_deploy_0004_set_card_field_default_values(&$sandbox) {
           \Drupal::logger('epa_core')->error("Error inserting record into $field_table table: " . $e->getMessage());
         }
       }
-    else {
-      // Record does exist but is null, run update.
-      try {
-        $database->update($field_table)->fields([
-          'field_flag_card_alignment_value' => 'default'
-        ])->execute();
-      }
-      catch (\Exception $e) {
-        \Drupal::logger('epa_core')->error("Error updating record into $field_table table: " . $e->getMessage());
+      else {
+        // Record does exist but is null, run update.
+        try {
+          $database->update($field_table)->fields([
+            'field_flag_card_alignment_value' => 'default'
+          ])->execute();
+        }
+        catch (\Exception $e) {
+          \Drupal::logger('epa_core')->error("Error updating record into $field_table table: " . $e->getMessage());
+        }
+
       }
+    }
+  }
+}
+
+/**
+ * Setting notification_opt_in flag for node author, author of the latest
+ * revision, or author of the current revision
+ */
+function epa_core_deploy_0005_set_watch_flag(&$sandbox) {
+  if (!isset($sandbox['total'])) {
+    // Get all nodes.
+    $nodes = \Drupal::database()->query(
+      "SELECT node.nid,
+         latest.revision_uid,
+         node.uid,
+         current.revision_uid
+FROM node_revision AS latest
+LEFT JOIN node_field_data AS node
+    ON latest.nid = node.nid
+LEFT JOIN node_revision AS current
+    ON node.vid = current.vid
+WHERE latest.vid IN
+    (SELECT MAX(vid)
+    FROM node_revision
+    GROUP BY  nid);")
+      ->fetchAll();
+
+    $sandbox['total'] = count($nodes);
+    $sandbox['current'] = 0;
+    \Drupal::logger('epa_core')
+      ->notice('Queueing ' . count($nodes) . ' nodes to be updated with notification_opt_in flag');
+  }
+
+  // Query 500 at a time for batch.
+  $nodes = \Drupal::database()->query(
+    "SELECT node.nid,
+         latest.revision_uid as latest_uid,
+         node.uid,
+         current.revision_uid
+FROM node_revision AS latest
+LEFT JOIN node_field_data AS node
+    ON latest.nid = node.nid
+LEFT JOIN node_revision AS current
+    ON node.vid = current.vid
+WHERE latest.vid IN
+    (SELECT MAX(vid)
+    FROM node_revision
+    GROUP BY  nid) LIMIT 500
+        OFFSET " . $sandbox['current'] . ";")
+    ->fetchAll();
```

Comment:
Your query doesn't set an order, so you'll get these records back in a non-deterministic fashion leading to the potential paging problems we spoke about.  I'd recommend ordering by node ID ASC.